### PR TITLE
Added url validation for starting with http, https

### PR DESCRIPTION
**Pull request by WingCode**
The validation (starting with http, https) is expected by the docstring but not enforced.
It might be helpful to fail early and fast since there might be some prior resource intensive steps before utilising the IQM client.

### CODE REVIEW

File: src/iqm_client/iqm_client.py [hunk 1]
```diff
@@ -245,6 +250,8 @@ def __init__(
             username: Optional[str] = None,
             api_key: Optional[str] = None
     ):
+        if not url.startswith(("http","https")):
```
Comment (hay-k) [inline src/iqm_client/iqm_client.py:253]:
```suggestion
        if not url.startswith(("http:", "https:")):
```

**Review by hay-k — CHANGES_REQUESTED** [review]
You will also need to add a unit test case to cover this new exception. Let me know if you need any help.

**Comment by hay-k** [comment]
@WingCode Thanks for your pull request. I have reviewed it and requested some changes above.

**Review by WingCode — COMMENTED** [review]
Addressed review comments. Fix changelog, IQM client, added  tests.

File: tests/test_iqm_client.py [hunk 2]
```diff
@@ -134,3 +134,17 @@ def test_user_warning_is_emitted_when_warnings_in_response(base_url, settings_di
...
+        client = IQMClient(base_url, settings_dict)
+    except Exception as exc:
+        assert False, f" IQMClient raised an exception {exc}"
+
+
+def test_base_url_is_invalid(settings_dict):
+    invalid_base_url = 'https//example.com'
+    with pytest.raises(ClientConfigurationError) as exc:
+        client = IQMClient(invalid_base_url, settings_dict)
+    assert f"The URL schema has to be http or https. Incorrect schema in URL: {invalid_base_url}" == str(exc.value)
```
Comment (hay-k) [inline tests/test_iqm_client.py:150]:
```suggestion
    assert f"The URL schema has to be http or https. Incorrect schema in URL: {invalid_base_url}" == str(exc.value)
    
```

File: tests/test_iqm_client.py [hunk 3]
```diff
@@ -134,3 +134,17 @@ def test_user_warning_is_emitted_when_warnings_in_response(base_url, settings_di
             .thenReturn(mock({'status_code': 200, 'text': json.dumps({'status': 'ready', 'warnings': [msg]})})):
         with pytest.warns(UserWarning, match=msg):
             client.get_run(existing_run)
+
+
+def test_base_url_is_valid(base_url, settings_dict):
+    try:
+        client = IQMClient(base_url, settings_dict)
+    except Exception as exc:
+        assert False, f" IQMClient raised an exception {exc}"
```
Comment (hay-k) [inline tests/test_iqm_client.py:143]: I think you can remove this test, since it is naturally covered by the other tests already.

**Review by hay-k — CHANGES_REQUESTED** [review]

**Review by hay-k — APPROVED** [review]

**Pull request merged**